numba/parfors/array_analysis.py

```diff
@@ -450,10 +451,9 @@ def _get_names(self, obj):
             return tuple(self._get_names(x)[0] for x in obj)
         elif isinstance(obj, int):
             return (obj,)
-        else:
-            raise NotImplementedError(
-                "ShapeEquivSet does not support {}".format(obj)
-            )
+        warnings.warn(errors.NumbaPerformanceWarning(
+            f"Ignoring untracked object type {type(obj)} in ShapeEquivSet"))
+        return ()
```

Comment:
We have to check all the uses of `_get_names` in the code to make sure this empty tuple output works as expected. For example, `get_equiv_const` uses `names[0]` for the output of `_get_names` which can fail for empty tuple. Also, some places may expect the length of this to be the same as the tuple.

Reply:
From what I see, the places that assume size of at least 1 are:
- get_equiv_const
- get_equiv_set
- A recursive call to `_get_names` with a tuple

To test the recursive call to `_get_names` you can probably use `((1.2, 1.1))`

I'm not sure how to test the equiv ones, but since they clearly break in the empty tuple case, we can probably replace `> 1` with `!= 1` without any consequences.